### PR TITLE
add support for bpf_attr type

### PR DESCRIPTION
**Pull request by sarahfujimori**
This patch allows us to collect name, program type, and instruction
count of eBPF programs when they are verified.

Signed-off-by: Sarah Fujimori <sarah.fujimori@isovalent.com>

### CODE REVIEW

**Review by tpapagian — COMMENTED** [review]
Thanks!

Some initial comments:
1. You should also commit the generated files (i.e. inside `api/` directory type `make` to generate grpc files and then in the top-level directory `make codegen && make generate`).
2. It seems that it lacks proper code formatting (i.e. `make format`).
3. In `crds/examples/` there are a couple of examples on how to use various types. It would be great if you can add an example there as well.
4. It would be also great if we can have a simple unit test for that (but this can be also done as a followup into a separate PR).

**Review by tpapagian — COMMENTED** [review]
There is an error in `checkpatch.pl` which requires a header similar to [this](https://github.com/cilium/tetragon/blob/391dc9777bb8c65f80685972987c781177e04bbf/bpf/process/bpf_execve_event.c#L1-L2) in `bpf/process/types/bpfattr.h`.

Other than that, this looks good to me.

File: api/v1/tetragon/tetragon.proto [hunk 1]
```diff
@@ -170,6 +176,7 @@ message KprobeArgument {
 	KprobeSock sock_arg = 9;
 	KprobeCred cred_arg = 10;
 	int64 long_arg = 11;
+    KprobeBpfAttr bpf_attr_arg = 12;
```
Comment (tpapagian) [inline api/v1/tetragon/tetragon.proto:179]: Please fix also this formatting issue. I think that our linters do not run on the `.proto` file.

**Comment by willfindlay** [comment]
We can ignore the checkpatch job for now, that warning is a false positive which I am fixing here https://github.com/cilium/tetragon/pull/258

**Comment by willfindlay** [comment]
@sarahfujimori Mind running `go mod tidy -compat=1.17 && go mod vendor` and rebasing? The test failure we are seeing here was also allegedly fixed in https://github.com/cilium/tetragon/pull/235 so a rebase on top of latest main should do the trick there too.

**Comment by jrfastab** [comment]
needs a rebase as well..

**Comment by sarahfujimori** [comment]
see #275

**Pull request closed without merging**